lib/trino-orc/src/main/java/io/trino/orc/metadata/statistics/ColumnStatistics.java

```diff
@@ -53,6 +54,7 @@ public ColumnStatistics(
             BooleanStatistics booleanStatistics,
             IntegerStatistics integerStatistics,
             DoubleStatistics doubleStatistics,
+            Long numberOfNanValues,
```

Comment:
Why not put it inside `DoubleStatistics`?

Reply:
Feels you coud keep min/max as null if there are NaNs but keep nans count inside object. Woudl that not work?

Reply:
That was my initial intention as well.
However the `DoubleStatisticsBuilder` doesn't allow dealing with NaNs

https://github.com/trinodb/trino/blob/fe608f2723842037ff620d612a706900e79c52c8/lib/trino-orc/src/main/java/io/trino/orc/metadata/statistics/DoubleStatisticsBuilder.java#L93-L99

cc @dain